alacritty/src/config/merge.rs

```diff
@@ -0,0 +1,85 @@
+//! Merge multiple configuration files together.
+
+use serde_yaml::mapping::Mapping;
+use serde_yaml::Value;
+
+/// Merge two serde structures.
+///
+/// This will take all values from `replacement` and use `base` whenever a value isn't present in
+/// `replacement`.
+pub fn merge(base: Value, replacement: Value) -> Value {
+    match (base, replacement) {
+        (Value::Sequence(mut base), Value::Sequence(mut replacement)) => {
+            base.append(&mut replacement);
+            Value::Sequence(base)
+        },
+        (Value::Mapping(base), Value::Mapping(replacement)) => {
+            Value::Mapping(merge_mapping(base, replacement))
+        },
+        (_, value) => value,
+    }
+}
+
+/// Merge two key/value mappings.
+fn merge_mapping(mut base: Mapping, replacement: Mapping) -> Mapping {
+    for (key, value) in replacement {
+        let value = match base.remove(&key) {
+            Some(base_value) => merge(base_value, value),
+            None => value,
+        };
+        base.insert(key, value);
+    }
+
+    base
+}
```

Comment:
I think merging them by including both makes the most sense in general, but I can see how it would create issues.

On the other hand, you might want to have global keybinding configuration and then add new bindings in your user config. So I feel like this would be beneficial.

Reply:
I mean that you'll have 2 bindings for the same key, meaning that you can't override global bindings with your user config resulting in a bit weird behavior.

Reply:
You can't *override* them, but you can still add to them. If you would just replace all of them, you couldn't have that anymore.

Reply:
Oh, maybe you're getting me a bit wrong. I want to perform substitution like we have when overrding default builtin bindings with the ones in a config, but with global/local config combination.